lib/react.js

```diff
@@ -230,6 +230,8 @@ declare module react {
   declare var Component: typeof React$Component;
   declare var PureComponent: typeof React$PureComponent;
   declare var Element: typeof React$Element;
+  
+  declare type Node = string | number | React$Element<any> | Array<Node>;
```

Comment:
This can have boolean/void/null in it as well. It can also be an Iterable of Node.

https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/isomorphic/ReactTypes.js#L18-L24

Although I was thinking that `Node` should not allow boolean/undefined/void those but a fragment should be able to contain them.